src/helm/proxy/clients/vision_language/open_flamingo_client.py

```diff
@@ -0,0 +1,125 @@
+from threading import Lock
+from typing import List
+
+import torch
+from huggingface_hub import hf_hub_download
+from helm.proxy.clients.vision_language.open_flamingo import create_model_and_transforms
+
+from helm.common.cache import CacheConfig
+from helm.common.images_utils import open_image
+from helm.common.gpu_utils import get_torch_device_name
+from helm.common.media_object import TEXT_TYPE
+from helm.common.optional_dependencies import handle_module_not_found_error
+from helm.common.request import Request, RequestResult, Sequence, Token
+from helm.common.request import wrap_request_time
+from helm.proxy.clients.client import CachingClient, generate_uid_for_multimodal_prompt
+
+try:
+    from PIL import Image
+except ModuleNotFoundError as e:
+    handle_module_not_found_error(e, ["images"])
+
+
+class OpenFlamingoClient(CachingClient):
+    """
+    OpenFlamingo is an open source implementation of DeepMind's Flamingo models.
+    https://huggingface.co/openflamingo/OpenFlamingo-9B-vitl-mpt7b
+    """
+
+    END_OF_CHUNK_TOKEN: str = "<|endofchunk|>"
+    IMAGE_TOKEN: str = "<image>"
+
+    _model_lock: Lock = Lock()
+
+    def __init__(self, cache_config: CacheConfig):
+        super().__init__(cache_config)
+        self._device: str = get_torch_device_name()
+        self._get_model()
+
+    def _get_model(self):
+        with self._model_lock:
+            self._model, self.image_processor, self.tokenizer = create_model_and_transforms(
+                clip_vision_encoder_path="ViT-L-14",
+                clip_vision_encoder_pretrained="openai",
+                lang_encoder_path="anas-awadalla/mpt-7b",
+                tokenizer_path="anas-awadalla/mpt-7b",
+                cross_attn_every_n_layers=4,
+            )
+            self.tokenizer.padding_side = "left"
+            checkpoint_path = hf_hub_download("openflamingo/OpenFlamingo-9B-vitl-mpt7b", "checkpoint.pt")
```

Comment:
Here you hard-coded the model which is not the HELM philosophy. We want to make this client reusable if other versions of Flamingo come out. You should pass a `model_name` to the `__init__` function and use it here.